pkg/kubectl/describe_test.go

```diff
@@ -513,11 +513,144 @@ func TestDescribeDeployment(t *testing.T) {
 		},
 	})
 	d := DeploymentDescriber{fake}
-	out, err := d.Describe("foo", "bar")
+	out, err := d.Describe("foo", "bar", DescriberSettings{ShowEvents: true})
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
 	if !strings.Contains(out, "bar") || !strings.Contains(out, "foo") {
 		t.Errorf("unexpected out: %s", out)
 	}
 }
+
+func TestDescribeEvents(t *testing.T) {
+
+	events := &api.EventList{
+		Items: []api.Event{
+			{
+				Source:         api.EventSource{Component: "kubelet"},
+				Message:        "Item 1",
+				FirstTimestamp: unversioned.NewTime(time.Date(2014, time.January, 15, 0, 0, 0, 0, time.UTC)),
+				LastTimestamp:  unversioned.NewTime(time.Date(2014, time.January, 15, 0, 0, 0, 0, time.UTC)),
+				Count:          1,
+				Type:           api.EventTypeNormal,
+			},
+		},
+	}
+
+	m := map[string]Describer{
+		"DaemonSetDescriber": &DaemonSetDescriber{
+			testclient.NewSimpleFake(&extensions.DaemonSet{
+				ObjectMeta: api.ObjectMeta{
+					Name:      "bar",
+					Namespace: "foo",
+				},
+			}, events),
+		},
+		"DeploymentDescriber": &DeploymentDescriber{
+			fake.NewSimpleClientset(&extensions.Deployment{
+				ObjectMeta: api.ObjectMeta{
+					Name:      "bar",
+					Namespace: "foo",
+				},
+			}, events),
+		},
+		"EndpointsDescriber": &EndpointsDescriber{
+			testclient.NewSimpleFake(&api.Endpoints{
+				ObjectMeta: api.ObjectMeta{
+					Name:      "bar",
+					Namespace: "foo",
+				},
+			}, events),
+		},
+		//"HorizontalPodAutoscalerDescriber": &HorizontalPodAutoscalerDescriber{
```

Comment:
I have change all the three describers into TODO note. Opened separate issue for them [1].

[1] https://github.com/kubernetes/kubernetes/issues/24731